sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -41,9 +43,22 @@ private[sql] class SharedState(val sparkContext: SparkContext) {
   val listener: SQLListener = createListenerAndUI(sparkContext)
 
   /**
+   * The base hadoop configuration which is shared among all spark sessions. It is based on the
+   * default hadoop configuration of Spark, with custom configurations inside `hive-site.xml`.
+   */
+  lazy val hadoopConf: Configuration = {
+    val conf = sparkContext.hadoopConfiguration
```

Comment:
Sorry. I missed this part. Can we create a new `Configuration` instead of calling `addResource` directly? So, this PR will not change the behavior of other components which use `sparkContext.hadoopConfiguration`.